hbase-server/src/main/java/org/apache/hadoop/hbase/master/replication/ReplicationPeerManager.java

```diff
@@ -493,12 +499,35 @@ private void checkConfiguredWALEntryFilters(ReplicationPeerConfig peerConfig)
     }
   }
 
-  private void checkClusterKey(String clusterKey) throws DoNotRetryIOException {
+  private void checkClusterKeyForHBaseInterClusterReplicationEndpoint(String clusterKey)
+    throws DoNotRetryIOException {
     try {
       ZKConfig.validateClusterKey(clusterKey);
     } catch (IOException e) {
       throw new DoNotRetryIOException("Invalid cluster key: " + clusterKey, e);
     }
+    String peerClusterId = "";
+    try {
+      Configuration otherConf = HBaseConfiguration.createClusterConf(conf, clusterKey);
+      ZKWatcher zkWatcher =
+        new ZKWatcher(otherConf, this + "check-peer-cluster-key", new Abortable() {
+          @Override public void abort(String why, Throwable e) {
+          }
+
+          @Override public boolean isAborted() {
+            return false;
+          }
+        });
+      peerClusterId = ZKClusterId.readClusterIdZNode(zkWatcher);
```

Comment:
ReplicationPeerManager is colocated with master, right? you can use master.getClusterId() without doing all this? (that is cached, so saves a ZK round trip).

Reply:
Not right. This fetch peer cluster id.

Reply:
See the changed code in  hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java